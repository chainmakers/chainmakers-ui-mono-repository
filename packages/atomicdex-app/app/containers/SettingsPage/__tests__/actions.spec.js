import {
  openApplicationDialog,
  closeApplicationDialog,
  openMM2Dialog,
  closeMM2Dialog,
  switchTab
} from '../actions';
import {
  APPLICATION_DIALOG_OPEN,
  APPLICATION_DIALOG_CLOSE,
  MM2_DIALOG_OPEN,
  MM2_DIALOG_CLOSE,
  TAB_SWITCH
} from '../constants';

describe('containers/SettingsPage/actions/openApplicationDialog', () => {
  it('should openApplicationDialog should create openApplicationDialog action', () => {
    expect(openApplicationDialog()).toMatchSnapshot();
  });

  it('should return the correct type', () => {
    const expectedResult = {
      type: APPLICATION_DIALOG_OPEN
    };

    expect(openApplicationDialog()).toEqual(expectedResult);
  });
});

describe('containers/SettingsPage/actions/closeApplicationDialog', () => {
  it('should closeApplicationDialog should create closeApplicationDialog action', () => {
    expect(closeApplicationDialog()).toMatchSnapshot();
  });

  it('should return the correct type', () => {
    const expectedResult = {
      type: APPLICATION_DIALOG_CLOSE
    };

    expect(closeApplicationDialog()).toEqual(expectedResult);
  });
});

describe('containers/SettingsPage/actions/openMM2Dialog', () => {
  it('should openMM2Dialog should create openMM2Dialog action', () => {
    expect(openMM2Dialog()).toMatchSnapshot();
  });

  it('should return the correct type', () => {
    const expectedResult = {
      type: MM2_DIALOG_OPEN
    };

    expect(openMM2Dialog()).toEqual(expectedResult);
  });
});

describe('containers/SettingsPage/actions/closeMM2Dialog', () => {
  it('should closeMM2Dialog should create closeMM2Dialog action', () => {
    expect(closeMM2Dialog()).toMatchSnapshot();
  });

  it('should return the correct type', () => {
    const expectedResult = {
      type: MM2_DIALOG_CLOSE
    };

    expect(closeMM2Dialog()).toEqual(expectedResult);
  });
});

describe('containers/SettingsPage/actions/switchTab', () => {
  it('should switchTab should create switchTab action', () => {
    expect(switchTab()).toMatchSnapshot();
  });

  it('should return the correct type', () => {
    const expectedResult = {
      type: TAB_SWITCH,
      payload: {
        tab: 0
      }
    };

    expect(switchTab()).toEqual(expectedResult);
  });

  it('should return the correct type and tab', () => {
    const expectedResult = {
      type: TAB_SWITCH,
      payload: {
        tab: 3
      }
    };

    expect(switchTab(3)).toEqual(expectedResult);
  });
});
