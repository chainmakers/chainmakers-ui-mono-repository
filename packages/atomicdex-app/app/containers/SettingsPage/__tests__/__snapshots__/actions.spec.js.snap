// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/SettingsPage/actions/closeApplicationDialog should closeApplicationDialog should create closeApplicationDialog action 1`] = `
Object {
  "type": "atomicapp/SettingsPage/APPLICATION_DIALOG_CLOSE",
}
`;

exports[`containers/SettingsPage/actions/closeMM2Dialog should closeMM2Dialog should create closeMM2Dialog action 1`] = `
Object {
  "type": "atomicapp/SettingsPage/MM2_DIALOG_CLOSE",
}
`;

exports[`containers/SettingsPage/actions/openApplicationDialog should openApplicationDialog should create openApplicationDialog action 1`] = `
Object {
  "type": "atomicapp/SettingsPage/APPLICATION_DIALOG_OPEN",
}
`;

exports[`containers/SettingsPage/actions/openMM2Dialog should openMM2Dialog should create openMM2Dialog action 1`] = `
Object {
  "type": "atomicapp/SettingsPage/MM2_DIALOG_OPEN",
}
`;

exports[`containers/SettingsPage/actions/switchTab should switchTab should create switchTab action 1`] = `
Object {
  "payload": Object {
    "tab": 0,
  },
  "type": undefined,
}
`;
